File: csharp/raindrops/Raindrops.cs
using System.Linq;
using System.Collections.Generic;

internal class Raindrops
{
  static Dictionary<int, string> factorMap = new Dictionary<int, string>() {
    { 3, "Pling" },
    { 5, "Plang" },
    { 7, "Plong" }
  };

  internal static string Convert(int number)
  {
    return string.Concat(
      factorMap
      .Where(factor => number % factor.Key == 0)
      .Select(factor => factor.Value)
      .DefaultIfEmpty(number.ToString())
      );
  }
}

File: fsharp/raindrops/solution/Raindrops.fs
module Raindrops
    let convert number = 
        let getSound state factor = 
            if number % (fst factor) = 0 then state + (snd factor) else state

        [(3, "Pling"); (5, "Plang"); (7, "Plong")]
        |> List.fold getSound ""
        |> function
            | "" -> string number
            | sounds -> sounds            
        